### PR TITLE
[WIP] theta-l integration

### PR DESCRIPTION
**Pull request by oksanaguba**
Provides changes needed to run theta-l within the model.

### CODE REVIEW

**Comment by oksanaguba** [comment]
The last commit ran on anvil for '${which}/cime/scripts/create_newcase -case $fol -res ne30_ne30 -mach anvil -compiler intel -compset FC5AV1C-L' and is running now for 5 more years. Need to clean up code more before review.

**Comment by oksanaguba** [comment]
unrelated to current issues: namelist mod uses model_theta_l to define default standalone values for theta-l.

**Comment by oksanaguba** [comment]
This commit was run for test
```
brr=~/acme-branch
master=~/acme-master
test=ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel
baselinefold=~/as/acme-tests

#generate baseline from master
#${master}/cime/scripts/create_test -g --baseline-root ${baselinefold} ${test}

#run test from branch
${brr}/cime/scripts/create_test -c --baseline-root ${baselinefold} -b master ${test}
```
and the output is all pass.

```
20181204_164010_xy17pl: 1 test
  ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel (Overall: PASS) details:
    PASS ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel CREATE_NEWCASE
    PASS ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel XML
    PASS ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel SETUP
    PASS ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel SHAREDLIB_BUILD time=406
    PASS ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel NLCOMP
    PASS ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel MODEL_BUILD time=516
    PASS ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel SUBMIT
    PASS ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel RUN time=66
    PASS ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel COMPARE_base_rest
    PASS ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel BASELINE master:
    PASS ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel TPUTCOMP
    PASS ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel MEMLEAK insuffiencient data for memleak test
    PASS ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel SHORT_TERM_ARCHIVER
```

File: components/cam/src/dynamics/se/dyn_comp.F90 [hunk 1]
```diff
@@ -210,43 +212,47 @@ subroutine dyn_init1(fh, NLFileName, dyn_in, dyn_out)
     ! Physics-grid will be defined later by phys_grid_init
     call define_cam_grids()
 
+    !why store ps0 in a temp?
```
Comment (oksanaguba) [inline components/cam/src/dynamics/se/dyn_comp.F90]: after more changes i don't think hybrid needs to be init-ed here.

**Comment by oksanaguba** [comment]
the test from yesterday, ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel, never computed diffs wrt master, so, there is a bug somewhere to locate and it is not clear which commit introduced it.

**Comment by oksanaguba** [comment]
The test now passes with this commit, but by some reason cupric does not run automatically, so I ran it by hand:
```
/home/ccsm-data/tools/cprnc /home/onguba/as/acme-tests/master/ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel/cpl.hi.0001-01-01-64800.nc run/ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel.G.20181204_144715_rbehv4.cpl.hi.0001-01-01-64800.nc
```
Identical files.

**Comment by oksanaguba** [comment]
list of things to do: 
1 change dss of FM back to FM(:,:,:,1:2) in stepon_run2 and probably in init
2 switch to #ifdef MODEL_THETA_L in stepon and in prim_advance_mod
3 get rid of dummies for the preqx model (convert_ routines)
4 get rid of ftype3

**Comment by oksanaguba** [comment]
this commit passes homme suite tests
```
100% tests passed, 0 tests failed out of 17
```

**Comment by oksanaguba** [comment]
Latest commit passed ERP_Ln9.ne4_ne4.FC5AV1C-L.anvil_intel .

**Comment by oksanaguba** [comment]
The issue reported today was about cubed_sphere_map, setting it to 0 in user_nl_cam got this resolved.

**Comment by oksanaguba** [comment]
for current commit these tests fail
```
The following tests FAILED:
         12 - thetah-test22 (Failed)
         13 - thetanh-test22 (Failed)
         14 - thetah-TC (Failed)
         15 - thetanh-TC (Failed)
         16 - thetanhwet-TC (Failed)
```
since many other things went into master by now, I think the easiest is to merge master into this branch and test again.

**Comment by rljacob** [comment]
From IG telecon:  Mark requested further changes.

**Pull request closed without merging**